src/utility.cpp

```diff
@@ -377,4 +377,37 @@ opentracing::expected<uint64_t> HexToUint64(opentracing::string_view s) {
 
   return result;
 }
+
+//--------------------------------------------------------------------------------------------------
+// ReadChunkHeader
+//--------------------------------------------------------------------------------------------------
+bool ReadChunkHeader(google::protobuf::io::ZeroCopyInputStream& stream,
+                     size_t& chunk_size) {
+  std::array<char, 16> digits;
```

Comment:
There's a bare `17` elsewhere in this PR. Could you make a static const somewhere and refer to it both here and there? Is the 17 case different from the 16 case? If so please add a comment.

Reply:
I'll change to write it in terms of numeric_limits.

There's a extra byte here because std::snprintf needs to write a null terminator; the other code has 16 because reading the number doesn't require you to have a null terminator (since it uses a string_view).